user-service/cmd/main.go

```diff
@@ -0,0 +1,157 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-redis/redis/v8"
+	"github.com/spf13/viper"
+	"github.com/winkedin/user-service/models"
+	"github.com/winkedin/user-service/services"
+	"gorm.io/driver/postgres"
+	"gorm.io/gorm"
+	"gorm.io/gorm/logger"
+)
+
+var (
+	configFilePath = flag.String("config", "", "absolute path to the config file")
+)
+
+var (
+	db  *gorm.DB
+	rdb *redis.Client
+	ctx = context.Background()
+	v   = viper.New()
+)
+
+func initConfig() {
+	v.SetConfigFile(*configFilePath)
+	err := v.ReadInConfig()
+	if err != nil {
+		log.Fatalf("Error reading config file, %s", err)
+	}
+}
+
+func initDB() {
+	dsn := fmt.Sprintf(
+		"host=%s user=%s password=%s dbname=%s port=%d sslmode=%s",
+		v.GetString("database.host"),
+		v.GetString("database.user"),
+		v.GetString("database.password"),
+		v.GetString("database.dbname"),
+		v.GetInt("database.port"),
+		v.GetString("database.sslmode"),
+	)
+	var err error
+	db, err = gorm.Open(postgres.Open(dsn), &gorm.Config{
+		Logger: logger.Default.LogMode(logger.Info),
+	})
+	if err != nil {
+		log.Fatal("Failed to connect to the database: ", err)
+	}
+	db.AutoMigrate(&models.User{})
+}
+
+func initRedis() {
+	rdb = redis.NewClient(&redis.Options{
+		Addr:     fmt.Sprintf("%s:%d", v.GetString("redis.host"), v.GetInt("redis.port")),
+		Password: v.GetString("redis.password"),
+		DB:       0,
+	})
+	_, err := rdb.Ping(ctx).Result()
+	if err != nil {
+		log.Fatal("Failed to connect to Redis: ", err)
+	}
+}
+
+func main() {
+	flag.Parse()
+	initConfig()
+	initDB()
+	initRedis()
+	defer func() {
+		if err := rdb.Close(); err != nil {
+			log.Println(fmt.Sprintf("%v", err))
+		}
+	}()
+	emailVerificationSVC := services.NewEmailVerificationService(rdb)
+	signupSVC := services.NewSignupService(db, rdb, emailVerificationSVC)
+	loginSvc := services.NewLoginService(db, rdb)
+	router := gin.Default()
+
+	router.POST("/signup", func(c *gin.Context) {

```

Comment:
Can we integrate swagger for APIs and have the routes defined in swagger and implementation in the service.